bin/templates/cordova/lib/build.js

```diff
@@ -422,11 +477,15 @@ function parseOpts(options, resolvedTarget) {
 module.exports.runClean = function(options) {
     var opts = parseOpts(options);
     var builder = builders[opts.buildMethod];
-    return builder.prepEnv()
+    return builder.prepEnv(opts)
     .then(function() {
-        return builder.clean(opts.extraArgs);
+        return builder.clean(opts);
     }).then(function() {
         shell.rm('-rf', path.join(ROOT, 'out'));
+    }).then(function() {
+        ['debug', 'release'].forEach(function(config) {
+            removeIfExists(path.join(ROOT, config + SIGNING_PROPERTIES));
```

Comment:
Good point. To delete these files the user has to explicitly run the `clean` script  from platforms\android\cordova. This made me think about other scenarios as well for CLI workflow:
1. cordova build <nopackageInfo>
2. cordova build <packageInfo>
3. cordova build <nopackageInfo>

The third invocation will use the pacakgeInfo from the previous invocation as I do not delete debug-signing.properties & release-signing.properies.

I can make this a bit more sophisticated to account for non-CLI workflows - detect if the file is auto-generated (It has the canned comment ("Do not modify this file -- YOUR CHANGES WILL BE ERASED")). If it is then I go ahead and delete it not only on `clean` but also when starting a build with no package info specified.